src/main/kotlin/com/chattriggers/ctjs/minecraft/wrappers/BossBars.kt

```diff
@@ -0,0 +1,255 @@
+package com.chattriggers.ctjs.minecraft.wrappers
+
+import com.chattriggers.ctjs.mixins.BossBarHudAccessor
+import com.chattriggers.ctjs.utils.asMixin
+import com.chattriggers.ctjs.utils.getOption
+import gg.essential.universal.wrappers.message.UTextComponent
+import net.minecraft.client.gui.hud.BossBarHud
+import net.minecraft.client.gui.hud.ClientBossBar
+import org.mozilla.javascript.NativeObject
+import java.util.UUID
+import net.minecraft.entity.boss.BossBar as MCBossBar
+
+object BossBars : CTWrapper<BossBarHud> {
+    override val mcValue: BossBarHud
+        get() = Client.getMinecraft().inGameHud.bossBarHud
+
+    /**
+     * Gets the list of currently shown [BossBar]s
+     *
+     * @return the currently displayed [BossBar]s
+     */
+    @JvmStatic
+    fun getBossBars(): List<BossBar> {
+        return toMC().asMixin<BossBarHudAccessor>().bossBars.values.map(::BossBar)
+    }
+
+    /**
+     * Gets all [BossBar]s with a given name
+     *
+     * @param name the name to match
+     * @return the [BossBar]s
+     */
+    @JvmStatic
+    fun getBossBarsByName(name: String): List<BossBar> {
+        return getBossBars().filter { it.getName() == name }
+    }
+
+    /**
+     * Adds a new [BossBar] to be displayed
+     *
+     * Takes a parameter with the following options:
+     * - name: The name to appear above the BossBar. Defaults to an empty string
+     * - percent: The percent full the BossBar is. Defaults to 1 (full health)
+     * - color: The color of the BossBar. Can be any [Color], but defaults to white
+     * - sections: The number of notches/sections to appear on the BossBar. Can be any [Style], but
+     *             defaults to 1 entire section
+     * - darkenSky: Whether the BossBar should darken the screen of the player. Defaults to false
+     * - thickenFog: Whether the BossBar should thicken the fog around the player. Defaults to false
+     *
+     * @param obj An options bag
+     *
+     * @return the [BossBar] for further modification
+     */
+    @JvmStatic
+    fun addBossBar(obj: NativeObject): BossBar {
+        val name = obj.getOption("name", "")
+        val percent = obj.getOption("percent", 1f).toFloat().coerceIn(0f..1f)
+        val color = Color.from(obj.getOption("color", Color.WHITE))
+        val style = Style.from(obj.getOption("sections", Style.ONE))
+        val shouldDarkenSky = obj.getOption("darkenSky", false).toBoolean()
+        val shouldThickenFog = obj.getOption("thickenFog", false).toBoolean()
+
+        val bossBar = ClientBossBar(
+            UUID.randomUUID(),
+            UTextComponent(name),
+            percent,
+            color.toMC(),
+            style.toMC(),
+            shouldDarkenSky,
+            false,
+            shouldThickenFog
+        )
+
+        toMC().asMixin<BossBarHudAccessor>().bossBars[UUID.randomUUID()] = bossBar

```

Comment:
Does it not matter that this UUID is different from the one above?

Reply:
Pretty sure it doesn't, but just to be sure I'll make them use the same one.